src/nemos/base_regressor.py

```diff
@@ -0,0 +1,418 @@
+
+        Instantiate the solver with the provided loss function, and return callable functions
+        that initialize the solver state, update the model parameters, and run the optimization.
+
+        This method creates a solver instance from jaxopt library, tailored to the specific loss
+        function and regularization approach defined by the Regularizer instance. It also handles
+        the proximal operator if required for the optimization method. The returned functions are
+        directly usable in optimization loops, simplifying the syntax by pre-setting
+        common arguments like regularization strength and other hyperparameters.
+
+        Parameters
+        ----------
+        loss :
+            The loss function to be optimized.
+
+        *args:
+            Positional arguments for the jaxopt `solver.run` method, e.g. the regularizing
+            strength for proximal gradient methods.
+
+        prox:
+            Optional, the proximal projection operator.
+
+        *kwargs:
+            Keyword arguments for the jaxopt `solver.run` method.
+
+        Returns
+        -------
+        :
+            A tuple containing three callable functions:
+            - solver_init_state: Function to initialize the solver's state, necessary before starting the optimization.
+            - solver_update: Function to perform a single update step in the optimization process,
+            returning new parameters and state.
+            - solver_run: Function to execute the optimization process, applying multiple updates until a
+            stopping criterion is met.
+        """
+        # final check that solver is valid for chosen regularizer
+        if self.solver_name not in self.regularizer.allowed_solvers:
+            raise ValueError(
+                f"The solver: {self.solver_name} is not allowed for "
+                f"{self._regularizer.__class__} regularizaration. Allowed solvers are "
+                f"{self._regularizer.allowed_solvers}."
+            )
+
+        # only use penalized loss if not using proximal gradient descent
+        if self.solver_name != "ProximalGradient":
+            loss = self.regularizer.penalized_loss(
+                self._predict_and_compute_loss, self.regularizer_strength
+            )
+        else:
+            loss = self._predict_and_compute_loss
+
+        # check that the loss is Callable
+        utils.assert_is_callable(loss, "loss")
+
+        # some parsing to make sure solver gets instantiated properly
+        if self.solver_name == "ProximalGradient":
+            self.solver_kwargs.update(prox=self.regularizer.get_proximal_operator())
+            # add self.regularizer_strength to args
+            args += (self.regularizer_strength,)
+        if self.solver_name == "LBFGSB":
+            if "bounds" not in kwargs.keys():
+                warnings.warn(
+                    "Bounds must be provided for LBFGSB. Reverting back to LBFGS solver."
+                )
```

Comment:
- [ ] I'd rather raise an exception here -- if someone's trying to use LBFGSB but not setting bounds, they've misunderstood something
- [ ] why is this checking `kwargs` and not `self.solver_kwargs`? if I'm understanding the code correctly, there's no way for the user to specify `kwargs` at this point
- [ ] the following fails: `nmo.glm.GLM(solver_name='LBFGSB', solver_kwargs={'bounds': (0,1)})`. I think that's because bounds must be passed at runtime, not initialization, but then, following my point above, I'm unclear how the user can do so (and I can't find an example in either our docs or tests)

Reply:
Resolved by dropping support for LBFGSB, will add support back if needed.